man/homectl.xml

```diff
@@ -506,6 +506,40 @@
         <xi:include href="version-info.xml" xpointer="v247"/></listitem>
       </varlistentry>
 
+    <varlistentry>
+        <term><option>--blob-directory=</option><replaceable>PATH</replaceable></term>
+        <term><option>-b</option> <replaceable>PATH</replaceable></term>
+        <term><option>--blob-directory=</option><replaceable>FILENAME</replaceable>=<replaceable>PATH</replaceable></term>
+        <term><option>-b</option> <replaceable>FILENAME</replaceable>=<replaceable>PATH</replaceable></term>
+
+        <listitem><para>Accepts either a directory path, or a file name followed by a file path. If just a
+        directory path is specified, then the user's entire blob directory is replaced the specified path.
+        Note that this replacement is performed before per-file manipulations are applied, which means these per-file
+        manipulations will be applied on top of the specified directory. If a filename and file path are specified, then
+        the single specified file in the blob directory will be overwritten with the specified path. If completely
+        blank, all changes to the blob directory up to this point are reset. If a filename is specified but the corresponding
+        file path is blank, that single file will be deleted from the blob directory. All changes are performed in temporary
+        copies of the specified files and directories, which means that the originals are not modified. See
+        <ulink url="https://systemd.io/USER_RECORD_BLOB_DIRS">User Record Blob Directories</ulink> for more information
+        about blob directories.</para>
```

Comment:
maybe we should not expose --blob-directory= at all, but instead --blob-extra= or so which would take a pair of colon separated name and path

so that:

* `--blob-extra=avatar:/path/to/avatar.png` would be pretty much equivalent to `--avatar=/path/to/avatar.png`

* And deletion would be done via `--blob-extra=avatar:` (or `--avatar=`), i.e. assigning an empty string.

And `--blob-extra=` without any argument at all could mean: delete all blobs?

or in other words, hide the fact that there's a dir behind any of this in homectl

Reply:
IDK, I like the idea of being able to prepare a blob directory and then tell `homectl` to just replace the dir wholesale.

`--blob-directory` is implemented pretty much exactly as you say in this comment, save for three differences:
* I'm using equals instead of colon to separate the name from the path. So `--blob-directory=avatar=/path/to/avatar.png` works (with `--avatar=/path/to/avatar.png` as a shortcut). I can change it to a colon if your prefer
* There's an additional mode, where you can pass `--blob-directory=/path/to/dir` and it'll just replace the whole blob directory with the contents of the specified dir

* Right now `--blob-directory=` just resets any blob-related flags, BUT I was thinking it should delete all blobs instead (see https://github.com/systemd/systemd/pull/30840#discussion_r1454470873 above). So I'll change it

Perhaps I can rename `--blob-directory` to just `--blobs` or so if you want, to de-emphasize that it's actually a directory under the hood but still leave room to replace the whole directory?

Reply:
I've renamed `--blob-directory` to just `--blob`, and made `--blob=` empty out the directory.

I didn't change `--blob=filename=/path/to/data` to `--blob=filename:/path/to/data` because flags with similar behavior seem to prefer the former syntax